python_hb_report/collector.py

```diff
@@ -0,0 +1,374 @@
+		i = 0
+
+		utillib.debug('looking for PE files in'+envir.PE_STATE_DIR)
+		flist = utillib.find_files(envir.PE_STATE_DIR.split())
+		grep_pro = subprocess.Popen(['grep','-v',"[.]last$"],stdin = subprocess.PIPE,stdout=subprocess.PIPE,stderr=subprocess.STDOUT)
+		flist = grep_pro.communicate(' '.join(flist))[0].split()
+		
+		if len(flist):
+			filename = utillib.basename(envir.PE_STATE_DIR)
+			pengine_dir = os.path.join(workdir,filename)
+			os.mkdir(pengine_dir)
+			for f in flist:
+#				os.symlink(f,os.path.join(pengine_dir,utillib.basename(f)))
+				os.symlink(f,pengine_dir)
+				utillib.do_command(['ln','-s',f,pengine_dir])
+				i = i + 1
+			utillib.debug('found '+str(i)+' pengine input files in '+envir.PE_STATE_DIR)
+
+		if i >= 20:
+			for f in flist:
+				if not self.skip_lvl(1):
+					path = os.path.join(workdir,utillib.basename(envir.PE_STATE_DIR))
+					path = os.path.join(path,utillib.basename(f))
+					self.pe2dot(path)
+		else:
+			utillib.debug('too many PE inputs to create dot files')
+
+	def touch_DC_if_dc(self):
+		dc = utillib.do_command(['crmadmin','-D'])
+		dc = dc.split()[len(dc.split()) - 1]
+		if self.WE == dc:
+			utillib.writefile(os.path.join(self.WORKDIR,'DC'),'')
+
+	def getbacktraces(self):
+		flist = []
+		bt_files = utillib.find_files(envir.CORES_DIRS)
+		for f in bt_files:
+			bf = utillib.basename(f)
+			bf_num  = utillib.do_command(['expr','match',bf,'core'])
+			if bf_num > 0:
+				flist.append(f)
+		if len(flist):
+			utillib.getbt(flist,os.path.join(self.WORKDIR,envir.BT_F))
+			utillib.debug('found basktraces: '+' '.join(flist))
+
+	def getconfigurations(self):
+		dest = self.WORKDIR	
+
+		for conf in envir.CONFIGURATIONS:
+			if os.path.isfile(conf):
+				shutil.copyfile(conf,os.path.join(dest,utillib.basename(conf)))
+			elif os.path.isdir(conf):
+				files = os.listdir(conf)
+				dst = os.path.join(self.WORKDIR,utillib.basename(conf))
+				os.mkdir(dst)
+				for f in files:
+					src = os.path.join(conf,f)
+					shutil.copyfile(src,os.path.join(dst,f))
+
+	def time_status(self):
+		output = os.path.join(self.WORKDIR,envir.TIME_F)
+
+		time_info = utillib.do_command(['date'])
+		time_info = time_info + utillib.do_command(['ntpdc','-pn'])
+		utillib.writefile(output,time_info)
+
+	def corosync_blackbox(self):
+		outf = os.path.join(self.WORKDIR,envir.COROSYNC_RECORDER_F)
+		from_time = envir.FROM_TIME
+		to_time = envir.TO_TIME
+
+		inpf = utillib.find_files(['/var/lib/corosync'])
+
+		if os.path.isfile(' '.join(inpf)):
+			blkbox_info = utillib.do_command(['corosync-blackbox'])
+			utillib.writefile(outf,blkbox_info)
+			utillib.do_command(['touch','-r',inpf,outf])
+
+	def getratraces(self):
+		i = 0
+		trace_dir = os.path.join(envir.HA_VARLIB,'trace_ra')
+		
+		if not os.path.isdir(trace_dir):
+			return False
+
+		utillib.debug('looking for RA trace files in '+trace_dir)
+		sed_pro = subprocess.Popen(['sed',"s,"+utillib.dirname(trace_dir)+"/,,g"],stdin = subprocess.PIPE,stdout = subprocess.PIPE)
```

Comment:
it seems utillib.do_command can do the job here, as it is wrapper of subprocess.Popen, and here has no extra operation.